Extensions/Spine/spineruntimeobject.ts

```diff
@@ -0,0 +1,215 @@
+      super.onDestroyFromScene(instanceContainer);
+      this._renderer.onDestroy();
+    }
+
+    setTimeScale(timeScale: float): void {
+      this._timeScale = timeScale;
+      this._renderer.updateTimeScale();
+    }
+
+    getTimeScale(): float {
+      return this._timeScale;
+    }
+
+    setScale(scale: float): void {
+      this._scale = scale;
+      this._renderer.updateScale();
+      this.invalidateHitboxes();
+    }
+
+    getScale(): float {
+      return this._scale;
+    }
+
+    setX(x: float): void {
+      super.setX(x);
+      this._renderer.updatePosition();
+    }
+
+    setY(y: float): void {
+      super.setY(y);
+      this._renderer.updatePosition();
+    }
+
+    setAngle(angle: float): void {
+      super.setAngle(angle);
+      this._renderer.updateAngle();
+    }
+
+    setOpacity(opacity: float): void {
+      this._opacity =
+        opacity < 0 ? 0 :
+        opacity > 255 ? 255 :
+        opacity;
+      this._renderer.updateOpacity();
+    }
+
+    getOpacity(): float {
+      return this._opacity;
+    }
+
+    getWidth(): float {
+      return this._renderer.getWidth();
+    }
+
+    getHeight(): float {
+      return this._renderer.getHeight();
+    }
+
+    setWidth(width: float): void {
+      this._renderer.setWidth(width);
+      this.invalidateHitboxes();
+    }
+
+    setHeight(height: float): void {
+      this._renderer.setHeight(height);
+      this.invalidateHitboxes();
+    }
+
+    setAnimationIndex(animationIndex: number) {
+      if (!this.isAnimationIndex(animationIndex)) { return; }
+
+      const animation = this._animations[animationIndex];
+      this._currentAnimationIndex = animationIndex;
+      this._renderer.setAnimation(animation.source, animation.loop);
+    }
+
+    setAnimationName(animationName: string) {
+      this.setAnimationIndex(this.getAnimationIndex(animationName));
+    }
+
+    getCurrentAnimationIndex(): number {
+      return this._currentAnimationIndex;
+    }
+
+    getAnimationName(): string {
+      return this.isAnimationIndex(this._currentAnimationIndex) ?
+        this._animations[this._currentAnimationIndex].name : '';
+    }
+
+    getAnimationIndex(animationName: string) {
+      return this._animations.findIndex(animation => animation.name === animationName);
+    }
+
+    isAnimationIndex(animationIndex: number) {
+      return (
+        Number.isInteger(animationIndex)
+        && animationIndex >= 0
+        && animationIndex < this._animations.length
+      );
+    }
+
+    isAnimationComplete(): boolean {
+      return this._renderer.isAnimationCompelete;
+    }
+
+    isCurrentAnimationName(name: string): boolean {
+      return this.getAnimationName() === name;
+    }
+
+    setIsUpdatable(isUpdatable: boolean): void {
+      this._renderer.isUpdatable = isUpdatable;
+    }
+
+    isUpdatable(): boolean {
+      return this._renderer.isUpdatable;
+    }
+  }
+
+  gdjs.registerObject(
+    'SpineObject::SpineObject',
+    // @ts-ignore
```

Comment:
Can be removed? 
gdjs.SpineRuntimeObject extends from gdjs.RuntimeObject

Reply:
done